File: test-suite/src/test/java/io/micronaut/docs/informer/ConfigMapInformerTest.java
package io.micronaut.docs.informer;

import io.kubernetes.client.openapi.models.V1ConfigMap;
import io.kubernetes.client.openapi.models.V1ObjectMeta;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ConfigMapInformerTest {

    private static V1ConfigMap configMap(String name) {
        return new V1ConfigMap().metadata(new V1ObjectMeta().name(name).namespace("default"));
    }

    @Test
    void testTheHandlerRecordsTheEvents() {
        ConfigMapInformer informer = new ConfigMapInformer();

        informer.onAdd(configMap("added"));
        informer.onUpdate(configMap("added"), configMap("updated"));
        informer.onDelete(configMap("deleted"), false);

        assertEquals(List.of("added"), informer.getAdded().stream().map(cm -> cm.getMetadata().getName()).toList());
        assertEquals(List.of("updated"), informer.getUpdated().stream().map(cm -> cm.getMetadata().getName()).toList());
        assertEquals(List.of("deleted"), informer.getDeleted().stream().map(cm -> cm.getMetadata().getName()).toList());
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/informer/ConfigMapInformerTest.kt
package io.micronaut.docs.informer

import io.kubernetes.client.openapi.models.V1ConfigMap
import io.kubernetes.client.openapi.models.V1ObjectMeta
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ConfigMapInformerTest {

    private fun configMap(name: String) = V1ConfigMap().metadata(V1ObjectMeta().name(name).namespace("default"))

    @Test
    fun testTheHandlerRecordsTheEvents() {
        val informer = ConfigMapInformer()

        informer.onAdd(configMap("added"))
        informer.onUpdate(configMap("added"), configMap("updated"))
        informer.onDelete(configMap("deleted"), false)

        assertEquals(listOf("added"), informer.added.map { it.metadata!!.name })
        assertEquals(listOf("updated"), informer.updated.map { it.metadata!!.name })
        assertEquals(listOf("deleted"), informer.deleted.map { it.metadata!!.name })
    }
}
